File: client/src/app/shared/Subject.ts
import { ICourse } from "./Course";
import { ISubjectGrade } from "./SubjectGrade";
import { ITeacher } from "./Teacher";

export class ISubject {
    subjectId: number;
    subjectName: string;
    subjectTitle: string;
    isMajor: boolean;
    courseId: number;
    subjectGrade: ISubjectGrade[];
    course: ICourse;
    teacherId: number;
    subjectTeacher: ITeacher;
    totalStudents: number;

}

File: Data/Entities/Subject.cs
using System.Collections.Generic;

namespace Magnifinance.Data.Entities
{
    public class Subject
    {
        public int SubjectId { get; set; }
        public string SubjectName { get; set; }
        public string SubjectTitle { get; set; }
        public bool IsMajor { get; set; }

        public int CourseId { get; set; }
        public List<SubjectGrade> SubjectGrade { get; set; }
        public Course Course { get; set; }

        public int TeacherId { get; set; }
        public Teacher SubjectTeacher { get; set; }
    }
}
